sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala

```diff
@@ -44,12 +44,18 @@ trait CodegenInterpretedPlanTest extends PlanTest {
     val codegenMode = CodegenObjectFactoryMode.CODEGEN_ONLY.toString
     val interpretedMode = CodegenObjectFactoryMode.NO_CODEGEN.toString
 
-    withSQLConf(SQLConf.CODEGEN_FACTORY_MODE.key -> codegenMode) {
-      super.test(testName + " (codegen path)", testTags: _*)(testFun)(pos)
```

Comment:
`super.test` sets the sql conf again?

Reply:
`withSQLConf` depends on `SQLConf.get`. The existing code cannot guarantee the configuration setting at line 47 at `super.test`'s execution time.

Reply:
`SQLConf` is thread-local. Does `super.test` runs in another thread by the SBT test runner?

Reply:
Here, we define these tests by using `super.test` instead of executing.

Reply:
When the derived classes define test cases with this `test`, two test cases are created but `withSQLConf` is not inside the defined tests.

Reply:
ah, now I got it! What a hidden bug 😂